cds_ils/importer/XMLRecordDump.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015-2018 CERN.
+#
+# cds-migrator-kit is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS Migrator Records loader."""
+import datetime
+
+from cds_dojson.marc21.fields.books.errors import ManualMigrationRequired, \
+    MissingRequiredField, UnexpectedValue
+from cds_dojson.marc21.utils import create_record
+from invenio_records.api import Record
+
+from cds_ils.importer import marc21
+from flask import current_app
+
+from cds_ils.importer.handlers import importer_exception_handler
+from cds_ils.importer.errors import LossyConversion
+from invenio_pidstore.resolver import Resolver
+
+
+class XMLRecordDump(object):
+
+    def __init__(self, data, source_type='marcxml', latest_only=False,
+                 pid_fetchers=None, dojson_model=marc21):
+        """Initialize class."""
+        self.resolver = Resolver(
+            pid_type='recid', object_type='rec', getter=Record.get_record)
+        self.data = data
+        self.source_type = source_type
+        self.latest_only = latest_only
+        self.dojson_model = dojson_model
+        self.revisions = None
+        self.pid_fetchers = pid_fetchers or []
+
+    def dump(self):
+        dt = datetime.datetime.utcnow()
+
+        exception_handlers = {
+            UnexpectedValue: importer_exception_handler,
+            MissingRequiredField: importer_exception_handler,
+            ManualMigrationRequired: importer_exception_handler,
+        }
+
+        marc_record = create_record(self.data)
+
+        try:
+
+            # MARCXML -> JSON fields translation
+            val = self.dojson_model.do(
+                marc_record, exception_handlers=exception_handlers
+            )
+            missing = self.dojson_model.missing(marc_record)
```

Comment:
can you maybe describe better, with comments or better naming what `missing` means?
Missing required fields? Missing something else?

Ok, from the log it looks like rules missing. Does it mean that there are fields in the input Marc that are taken into account in the rules? I would just be a little bit more explicit.

Reply:
this naming is coming from cds-dojson, can't refactor it now since cds-videos is using it too, but I will leave a comment here